internal/usecases/liquidity_provider/summaries.go

```diff
@@ -0,0 +1,245 @@
+package liquidity_provider
+
+import (
+	"context"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/rsksmart/liquidity-provider-server/internal/entities"
+	"github.com/rsksmart/liquidity-provider-server/internal/entities/quote"
+)
+
+type SummariesResponse struct {
+	PeginSummary  SummaryData `json:"peginSummary"`
+	PegoutSummary SummaryData `json:"pegoutSummary"`
+}
+
+type SummaryData struct {
+	TotalQuotesCount          int64  `json:"totalQuotesCount"`
+	AcceptedQuotesCount       int64  `json:"acceptedQuotesCount"`
+	TotalQuotedAmount         string `json:"totalQuotedAmount"`
+	TotalAcceptedQuotedAmount string `json:"totalAcceptedQuotedAmount"`
+	TotalFeesCollected        string `json:"totalFeesCollected"`
+	RefundedQuotesCount       int64  `json:"refundedQuotesCount"`
+	TotalPenaltyAmount        string `json:"totalPenaltyAmount"`
+	LpEarnings                string `json:"lpEarnings"`
+}
+
+type SummariesUseCase struct {
+	peginRepo  quote.PeginQuoteRepository
+	pegoutRepo quote.PegoutQuoteRepository
+}
+
+func NewSummariesUseCase(
+	peginRepo quote.PeginQuoteRepository,
+	pegoutRepo quote.PegoutQuoteRepository,
+) *SummariesUseCase {
+	return &SummariesUseCase{
+		peginRepo:  peginRepo,
+		pegoutRepo: pegoutRepo,
+	}
+}
+
+func (u *SummariesUseCase) Run(ctx context.Context, startDate, endDate time.Time) (SummariesResponse, error) {
+	log.Printf("Running summaries from %s to %s", startDate.Format("2006-01-02"), endDate.Format("2006-01-02"))
+
+	peginData, err := u.aggregatePeginData(ctx, startDate, endDate)
+	if err != nil {
+		log.Printf("Error aggregating pegin data: %v", err)
+		return SummariesResponse{}, err
+	}
+
+	pegoutData, err := u.aggregatePegoutData(ctx, startDate, endDate)
+	if err != nil {
+		log.Printf("Error aggregating pegout data: %v", err)
+		return SummariesResponse{}, err
+	}
+
+	return SummariesResponse{
+		PeginSummary:  peginData,
+		PegoutSummary: pegoutData,
+	}, nil
+}
+
+func (u *SummariesUseCase) aggregatePeginData(ctx context.Context, startDate, endDate time.Time) (SummaryData, error) { //nolint:funlen,cyclop
+	var data SummaryData
+	quotes, retainedQuotes, err := u.peginRepo.ListQuotesByDateRange(ctx, startDate, endDate)
+	if err != nil {
+		log.Printf("Error listing pegin quotes: %v", err)
+		return data, err
+	}
+	var totalAmount = entities.NewWei(0)
+	var acceptedTotalAmount = entities.NewWei(0)
+	var totalFees = entities.NewWei(0)
+	var callFees = entities.NewWei(0)
+	var totalPenalty = entities.NewWei(0)
+	
+	processedQuotes := make(map[string]*quote.PeginQuote)
+	uniqueQuoteHashes := make(map[string]bool)
+	
+	for i := range quotes {
+		quoteHash := strconv.FormatInt(quotes[i].Nonce, 10)
+		uniqueQuoteHashes[quoteHash] = true
```

Comment:
all the quote hashes are unique is part of the business logic of the protocol

Reply:
if you're concerned about duplicated entries in the DB you can define a unique index